components/org.wso2.carbon.identity.conditional.auth.functions.elk/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/elk/HttpClientManager.java

```diff
@@ -0,0 +1,227 @@
+
+            RequestConfig config = createRequestConfig(tenantDomain);
+
+            HttpAsyncClientBuilder httpClientBuilder = HttpAsyncClients.custom().setDefaultRequestConfig(config);
+
+            addSslContext(httpClientBuilder, tenantDomain);
+            httpClientBuilder.setConnectionManager(poolingHttpClientConnectionManager);
+
+            client = httpClientBuilder.build();
+            client.start();
+            clientMap.put(tenantId, client);
+        }
+
+        return client;
+    }
+
+    private RequestConfig createRequestConfig(String tenantDomain) {
+
+        int defaultTimeout = 5000;
+        String connectionTimeoutString = null;
+        String readTimeoutString = null;
+        String connectionRequestTimeoutString = null;
+        try {
+            connectionTimeoutString = CommonUtils.getConnectorConfig(ElasticAnalyticsEngineConfigImpl
+                    .HTTP_CONNECTION_TIMEOUT, tenantDomain);
+        } catch (IdentityEventException e) {
+            // Ignore. If there was error while getting the property, continue with default value.
+        }
+        try {
+            readTimeoutString = CommonUtils.getConnectorConfig(ElasticAnalyticsEngineConfigImpl
+                    .HTTP_READ_TIMEOUT, tenantDomain);
+        } catch (IdentityEventException e) {
+            // Ignore. If there was error while getting the property, continue with default value.
+        }
+        try {
+            connectionRequestTimeoutString = CommonUtils.getConnectorConfig(ElasticAnalyticsEngineConfigImpl
+                    .HTTP_CONNECTION_REQUEST_TIMEOUT, tenantDomain);
+        } catch (IdentityEventException e) {
+            // Ignore. If there was error while getting the property, continue with default value.
+        }
+
+        int connectionTimeout = defaultTimeout;
+        int readTimeout = defaultTimeout;
+        int connectionRequestTimeout = defaultTimeout;
+
+        if (connectionTimeoutString != null) {
+            try {
+                connectionTimeout = Integer.parseInt(connectionTimeoutString);
+            } catch (NumberFormatException e) {
+                LOG.error("Error while parsing connection timeout : " + connectionTimeoutString, e);
+            }
+        }
+        if (readTimeoutString != null) {
+            try {
+                readTimeout = Integer.parseInt(readTimeoutString);
+            } catch (NumberFormatException e) {
+                LOG.error("Error while parsing read timeout : " + connectionTimeoutString, e);
+            }
+        }
+        if (connectionRequestTimeoutString != null) {
+            try {
+                connectionRequestTimeout = Integer.parseInt(connectionRequestTimeoutString);
+            } catch (NumberFormatException e) {
+                LOG.error("Error while parsing connection request timeout : " + connectionTimeoutString, e);
+            }
```

Comment:
Is the variable in these error messages correct? Also what happens if we sent null for each variable? Is it handled properly from the other end?

Reply:
Fixed with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/111